### DISCUSSION TITLE
Endless retries of service?

### DISCUSSION THREAD

**Discussion by hongkongkiwi** [finit-project/finit#242] [Q&A] [answered]
I see in the documentation that I can use retries:0 to disable retrying and retries:NUM, but in my application I have a need for unlimited retries and it never goes to a "crashed" state.

Can I request a small addition to add retries:-1 to disable a service being marked as crashed?

Also, whats the maximum number that can be specified for this? (I didn't see a max listed in docs).

Example of this is an LTE dialer which exits the script if it fails. So I don't want it to ever stop retrying to connect to LTE.

**Comment by troglobit** [comment, marked as answer]
I think you mean `restart:NUM`, right? Its MAX value is an `unsigned char`, so 255.

You can use secret option `respawn` as I've used in the `start-kill-service.sh` test case.  It's been undocumented for a reason, but maybe it's time for it to be opened up to the public since there are valid use-cases for it.

**Reply by hongkongkiwi** [reply]
Yes, restart that's it :) respawn is what I need, great.

**Reply by troglobit** [reply]
Awesome! I'll make sure to update the docs :+1:
